common/ferrostar/Cargo.toml

```diff
@@ -15,23 +15,33 @@ rust-version.workspace = true
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
+[features]
+alloc = []
+std = ["alloc", "serde_json/std", "proptest/std"]
+default = ["std", "uniffi"]
+wasm_js = ["std", "getrandom/js", "wasm-bindgen", "web-time"]
+
 [dependencies]
 geo = "0.28.0"
-polyline = "0.10.0"
+polyline = "0.11.0"
 serde = { version = "1.0.162", features = ["derive"] }
-serde_json = "1.0.96"
+serde_json = { version = "1.0.117", default-features = false }
 thiserror = "1.0.40"
-uniffi.workspace = true
+uniffi = { workspace = true, optional = true }
 uuid = { version = "1.8.0", features = ["v4"] }
+getrandom = { version = "0.2.15", optional = true }
+wasm-bindgen = { version = "0.2.92", optional = true }
+web-time = { version = "1.1.0", optional = true }
+wee_alloc = { version = "0.4.5", optional = true }
```

Comment:
Ooh good catch! I didn't realize it was unmaintained or had security issues.

And yes, I think we can probably use the default allocator. I added this while on a plane without WiFi when I mistakenly thought that wasm32-unknown-unknown was a no_std-only target. Turns out std is supported but a few things like time will panic and randomness needs JS.

Reply:
And another pain of wasm targets with std is artifact may be big, which can drastically degrade the user experiences. So we may still keep the wasm target with no_std. And for the same purpose, I think we may want to enable LTO/strip symbols in release builds in Cargo configs.

Reply:
Yeah, we can compare sizes once we actually care; bit of a premature optimization otherwise to remove std. All of the work is done for no_std wasm32-unknown-unknown except for an allocator.

LTO makes sense, but I'm not sold on stripping symbols. That negatively affects the ability of developers to debug issues in or profile production apps accurately, where you always want to ship a release build. Perhaps we distribute two variants though for the oddly space conscious mobile dev. Embedded apps will not use a pre compiled lib and will set their own build flags to strip if they want to on the final application.

Reply:
Resolved in latest commits.